libr/core/anal_tp.c

```diff
@@ -73,17 +73,10 @@ static void var_rename(RAnal *anal, RAnalVar *v, const char *name, ut64 addr) {
 	r_anal_var_rename (anal, fcn->addr, 1, v->kind, v->name, name, false);
 }
 
-static void var_retype(RAnal *anal, RAnalVar *var, const char *vname, char *type, ut64 addr, bool ref, bool pfx) {
-	if (!type || !var) {
-		return;
-	}
-	char *trim = type;
-	r_str_trim (trim);
```

Comment:
Why did you remove this? I think it is necessary in some cases. You should use `r_str_trim_dup (trim)` and free the resulting string at the end of the function.

Reply:
in which cases it is necessary? the argument is char * so it can be modified, and we should change it to const char *, but we should also not trim anything from that string because it shouldnt be receiving a type with trailing spaces for example...